src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala

```diff
@@ -107,19 +107,33 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
     response.transactionResponse shouldBe Some(expectedTxResponse)
   }
 
-  //FIXME: This test should be changed once the pending block doesn't equal to the latest one
-  it should "answer eth_getBlockByNumber with the correct parameters in None when the pending block is requested" in new TestSetup {
-    val bestBlockNumber = 10
+  it should "answer eth_getBlockByNumber with the correct block when the pending block is requested" in new TestSetup {
+
+    (blockGenerator.getPending _).expects().returns(Some(PendingBlock(blockToRequest, Nil)))
 
-    blockchain.save(blockToRequest.copy(header = blockToRequest.header.copy(number = bestBlockNumber)))
+    val request = BlockByNumberRequest(BlockParam.Pending, fullTxs = true)
+    val response = Await.result(ethService.getBlockByNumber(request), Duration.Inf).right.get
```

Comment:
Can we use `.futureValue` instead of `Await`? At any rate, I'm not fond of `Duration.Inf`